client/emojis.lua

```diff
@@ -0,0 +1,194 @@
+    ['cry'] = '😢',
+    ['sob'] = '😭',
+    ['sweat'] = '😓',
+    ['disappointed_relieved'] = '😥',
+    ['cold_sweat'] = '😰',
+    ['fearful'] = '😨',
+    ['anguished'] = '😧',
+    ['frowning_open_mouth'] = '😦',
+    ['triumph'] = '😤',
+    ['angry'] = '😠',
+    ['rage'] = '😡',
+    ['cursing'] = '🤬',
+    ['flushed'] = '😳',
+    ['hushed'] = '😯',
+    ['open_mouth'] = '😮',
+    ['astonished'] = '😲',
+    ['peeking'] = '🫣',
+    ['hand_over_mouth'] = '🫢',
+    ['exploding_head'] = '🤯',
+    ['scream'] = '😱',
+    ['grimacing'] = '😬',
+    ['exhaling'] = '😮💨',
+    ['dizzy'] = '😵',
+    ['dizzy_spiral'] = '😵💫',
+    ['melting'] = '🫠',
+    ['shaking'] = '🫨',
+    ['yawning'] = '🥱',
+    ['sleepy'] = '😪',
+    ['sleeping'] = '😴',
+    ['holding_tears'] = '🥹',
+    ['smiling_tear'] = '🥲',
+    ['heart'] = '❤️',
+    ['poultry_leg'] = '🍗',
+    ['potable_water'] = '🚰',
+    ['nauseated'] = '🤢',
+    ['vomiting'] = '🤮',
+    ['sneezing'] = '🤧',
+    ['monocle'] = '🧐',
+    ['skull'] = '💀',
+    ['poop'] = '💩',
+    ['skull_crossbones'] = '☠️',
+    ['bone'] = '🦴',
+    ['paw_prints'] = '🐾',
+    ['injured'] = '🤕',
+    ['no_pedestrians'] = '🚷',
+    ['toilet'] = '🚽',
+    ['ear'] = '👂',
+    ['thumbs_up'] = '👍',
+    ['thumbs_down'] = '👎',
+}
+
+-- Function to draw text in 3D space
+local function DrawText3D(coords, text)
+    local camCoords = GetGameplayCamCoord()
+    local dist = #(coords - camCoords)
+    local scale = 300 / (GetGameplayCamFov() * dist)
+
+    SetTextColour(230, 230, 230, 255)
+    SetTextScale(0.0, 0.5 * scale)
+    SetTextDropshadow(0, 0, 0, 0, 55)
+    SetTextDropShadow()
+    SetTextCentre(true)
+
+    BeginTextCommandDisplayText("STRING")
+    AddTextComponentSubstringPlayerName(text)
+    SetDrawOrigin(coords, 0)
+    EndTextCommandDisplayText(0.0, 0.0)
+    ClearDrawOrigin()
+end
+
+-- Function to display emoji above the ped's head
+local function DisplayEmoji(ped, emoji)
+    local playerPed = PlayerPedId()
+    local playerCoords = GetEntityCoords(playerPed)
+    local pedCoords = GetEntityCoords(ped)
+    local dist = #(playerCoords - pedCoords)
+
+    if dist <= 25 then
+        emojiDisplaying[ped] = (emojiDisplaying[ped] or 0) + 1
+        local isDisplaying = true
+
+        CreateThread(function()
+            Wait(5000) -- Display time in milliseconds
+            isDisplaying = false
+        end)
+
+        local offset = 1.0 + emojiDisplaying[ped] * 0.1
+        while isDisplaying do
+            if HasEntityClearLosToEntity(playerPed, ped, 17) then
+                local x, y, z = table.unpack(GetEntityCoords(ped))
+                z = z + offset
+                DrawText3D(vector3(x, y, z), emoji)
+            end
+            Wait(0)
+        end
+
+        emojiDisplaying[ped] = emojiDisplaying[ped] - 1
+    end
+end
+
+-- State bag change handler for emoji display (replaces old event-based system)
+AddStateBagChangeHandler('emoji', '', function(bagName, key, value, _unused, replicated)
```

Comment:
This handler implies only one emoji at t a time, yet the display function implies emoji stacking. I'll wait your clarification on the intended behavior.